File: Cryptography/Caesar Cipher/caesar.js
function caesarCipherEncrypt(plaintext, shift) {
  let ciphertext = "";

  for (let i = 0; i < plaintext.length; i++) {
    let char = plaintext[i];

    // Menggeser karakter huruf kapital
    if (char.match(/[A-Z]/)) {
      let code = ((char.charCodeAt() - 65 + shift) % 26) + 65;
      char = String.fromCharCode(code);
    }
    // Menggeser karakter huruf kecil
    else if (char.match(/[a-z]/)) {
      let code = ((char.charCodeAt() - 97 + shift) % 26) + 97;
      char = String.fromCharCode(code);
    }

    ciphertext += char;
  }

  return ciphertext;
}

let plaintext = prompt("Masukkan teks: ");
let shift = parseInt(prompt("Masukkan jumlah pergeseran: "));

let ciphertext = caesarCipherEncrypt(plaintext, shift);

console.log("Hasil Enkripsi: " + ciphertext);

File: Cryptography/Caesar Cipher/caesar.cpp
#include <iostream>
#include <string>

std::string caesarCipherEncrypt(std::string plaintext, int shift) {
    std::string ciphertext = "";
    
    for (char& c : plaintext) {
        // Menggeser karakter huruf kapital
        if (std::isupper(c)) {
            c = (c - 'A' + shift) % 26 + 'A';
        }
        // Menggeser karakter huruf kecil
        else if (std::islower(c)) {
            c = (c - 'a' + shift) % 26 + 'a';
        }
        
        ciphertext += c;
    }
    
    return ciphertext;
}

int main() {
    std::string plaintext;
    int shift;
    
    std::cout << "Masukkan teks: ";
    std::getline(std::cin, plaintext);
    
    std::cout << "Masukkan jumlah pergeseran: ";
    std::cin >> shift;
    
    std::string ciphertext = caesarCipherEncrypt(plaintext, shift);
    
    std::cout << "Hasil Enkripsi: " << ciphertext << std::endl;
    
    return 0;
}
